### PR TITLE
SSL/TLS Support

### PR DESCRIPTION
**Pull request by senseibaka**
Implements #98 
PR not ready to merge yet.

### CODE REVIEW

**Comment by c-j-hughes** [comment]
This is ready to merge. Please can you review @senseibaka ?

File: src/AdoNetCore.AseClient/Internal/TokenSendStream.cs [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                        // Set the header bytes to describe what is being transmitted
+                        header[0] = (byte)_writeBufferType;
+                        header[1] = (byte)(BufferStatus.TDS_BUFSTAT_EOM | _writeBufferStatus);
+                        header[2] = (byte)(header.Length >> 8);
+                        header[3] = (byte)header.Length;
+
+                        DumpBytes(header, header.Length);
+
+                        _innerStream.Write(header, 0, headerLength);
+#if ENABLE_ARRAY_POOL
```
Comment (senseibaka) [inline src/AdoNetCore.AseClient/Internal/TokenSendStream.cs]: Is this in the wrong spot? shouldn't it be after the `}`?

Comment (senseibaka) [reply src/AdoNetCore.AseClient/Internal/TokenSendStream.cs]: Oh I see, the `try {` is in an equivalent block

File: test/AdoNetCore.AseClient.Tests/Integration/Connection/TlsConnectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
+using System;
+using System.IO;
+using Dapper;
+using NUnit.Framework;
+
+namespace AdoNetCore.AseClient.Tests.Integration.Connection
+{
+    [TestFixture]
```
Comment (senseibaka) [inline test/AdoNetCore.AseClient.Tests/Integration/Connection/TlsConnectionTests.cs]: I'd probably mark this fixture as `Explicit = true`

Comment (c-j-hughes) [reply test/AdoNetCore.AseClient.Tests/Integration/Connection/TlsConnectionTests.cs]: The test is explicit in a sense - it only runs if the test suite is configured for with a TLS endpoint.

**Pull request merged**